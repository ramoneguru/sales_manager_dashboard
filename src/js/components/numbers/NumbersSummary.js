/**
 *
 * @author Drew Robinson (hello@drewrobinson.com)
 * @version 0.0.1
 * @param props
 * @exports NumbersSummary Class
 */

import React from 'react';
import SummaryItem from './SummaryItem';
import PropTypes from 'prop-types';

/**
 * Reresents summary for sales teams number of activities
 * @param props
 * @returns {XML}
 * @constructor
 * @desc prefer normal functions (not arrow functions) for stateless/refless components over over classes
 */
function NumbersSummary( props ) {
  let summaryItems = Object.keys(props.salesReps).map( key => {
    let data = props.activityNumbers.filter(function(entity){
      return parseInt(entity.repId) === parseInt(key);
    });

    return <SummaryItem key={ key } name={ props.salesReps[key].name } data={data} avatar={ props.salesReps[key].avatar } />
  });
  
  return (
    <div className="rep-summary">
      <div className="rep-summary__header">
        <div>
          <h3>Sales Rep.</h3>
        </div>
        <div>
          <h3>Connect Rate %</h3>
        </div>
      </div>
      {summaryItems}
    </div>
  )
}

NumbersSummary.propTypes =  {
  'activityNumbers' : PropTypes.array.isRequired,
  'salesReps'       : PropTypes.object.isRequired
}

NumbersSummary.defaultProps = {
  'activityNumbers' : [],
  'salesReps'       : {}
};

export default NumbersSummary;
