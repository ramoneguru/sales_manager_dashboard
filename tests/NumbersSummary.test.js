import React from 'react';
import { shallow, render, mount } from 'enzyme';
import NumbersSummary from '../src/js/components/numbers/NumbersSummary';


describe('NumbersSummary component ', function(){
  "use strict";

  let activityNumbers, salesReps, summary;

  beforeEach(function(){
    activityNumbers = [{
      "repId": 85719831,
      "30D": [ 1800, 1400, 800, 500 ],
      "90D": [ 5400, 3800, 3200, 1700 ],
      "12M": [ 21600, 18300, 13500, 1200 ]
    },
      {
        "repId": 85719832,
        "30D": [ 1800, 1400, 800, 500 ],
        "90D": [ 5400, 3800, 3200, 1700 ],
        "12M": [ 21600, 18300, 13500, 1200 ]
      }]
    salesReps = {
      "85719831": {
        "name": "Marie Curie",
        "avatar":"85719831.jpeg",
        "report":{
          "dials":3844,
          "connects":333,
          "opps":11,
          "deals":1,
          "dialConnectRatio":[12,1],
          "dialOppRatio":[349,1],
          "oppDealRatio":[11,1],
          "dialDealRatio":[3844,1]
        }
      },
      "85719832": {
        "name": "Marie Curie",
        "avatar":"85719831.jpeg",
        "report":{
          "dials":3844,
          "connects":333,
          "opps":11,
          "deals":1,
          "dialConnectRatio":[12,1],
          "dialOppRatio":[349,1],
          "oppDealRatio":[11,1],
          "dialDealRatio":[3844,1]
        }
      }
    }

    summary = shallow(
      <NumbersSummary activityNumbers={ activityNumbers } salesReps={ salesReps } />
    );
  });

  it('should render correctly', () => {
    expect(summary).toMatchSnapshot()
  });

  it('should contain one summary item per sales rep', () => {
    const items = summary.find('SummaryItem')
    expect(items.length).not.toBe(1)
    expect(items.length).toBe(2)
  });

});
