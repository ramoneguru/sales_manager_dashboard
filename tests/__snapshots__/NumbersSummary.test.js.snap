// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumbersSummary component  should render correctly 1`] = `
<div
  className="rep-summary"
>
  <div
    className="rep-summary__header"
  >
    <div>
      <h3>
        Sales Rep.
      </h3>
    </div>
    <div>
      <h3>
        Connect Rate %
      </h3>
    </div>
  </div>
  <SummaryItem
    avatar="85719831.jpeg"
    data={
      Array [
        Object {
          "12M": Array [
            21600,
            18300,
            13500,
            1200,
          ],
          "30D": Array [
            1800,
            1400,
            800,
            500,
          ],
          "90D": Array [
            5400,
            3800,
            3200,
            1700,
          ],
          "repId": 85719831,
        },
      ]
    }
    name="Marie Curie"
  />
  <SummaryItem
    avatar="85719831.jpeg"
    data={
      Array [
        Object {
          "12M": Array [
            21600,
            18300,
            13500,
            1200,
          ],
          "30D": Array [
            1800,
            1400,
            800,
            500,
          ],
          "90D": Array [
            5400,
            3800,
            3200,
            1700,
          ],
          "repId": 85719832,
        },
      ]
    }
    name="Marie Curie"
  />
</div>
`;
